src/operator/tensor/dot.cc

```diff
@@ -51,12 +150,16 @@ NNVM_REGISTER_OP(dot)
     dot(x,y)[0,0,1,1] = 0
     sum(x[0,0,:]*y[:,1,1]) = 0
 
-The storage type of ``dot`` output depends on storage types of inputs and transpose options:
+The storage type of ``dot`` output depends on storage types of inputs, transpose options and given
+hint for output storage type:
 
+Implemented sprase operations include:
 - dot(csr, default) = default
 - dot(csr.T, default) = row_sparse
 - dot(csr, row_sparse) = default
-- dot(default, csr) = csr
+- dot(default, csr) = csr on CPU only
+- dot(default, csr) = dense on GPU only
```

Comment:
Could you show the output storage with specific values of `forward_stype_hint`? e.g.

dot(csr.T, dense) = row_sparse
dot(csr.T, dense, forward_stype_hint='default') = default
dot(default, csr, forward_stype_hint='default') = default (GPU only)

What happens if someone uses dot(dense, dense, forward_stype_hint='csr')?